src/main/java/com/raoulvdberge/refinedstorage/apiimpl/autocrafting/CraftingPattern.java

```diff
@@ -273,7 +273,28 @@ public String toString() {
             ", byproducts=" + byproducts +
             '}';
     }
-
+    
+    @Override
+    public boolean equals (Object obj) {
+    	return this==obj || (obj instanceof ICraftingPattern && this.alike((ICraftingPattern)obj));
+    }
+    
+    Integer hashCode;
+    @Override
+    public int hashCode() {
+    	if(hashCode==null){
+    		hashCode=0;
+    		for(ItemStack mItemStack : this.getOutputs()){
+    			int itemHashCode=0;
+    			itemHashCode=mItemStack.getCount();
+    			itemHashCode=itemHashCode*31+mItemStack.getItem().hashCode();
+    			itemHashCode=itemHashCode*31+mItemStack.getItemDamage();
```

Comment:
Might want to include the hash of the NBT data, since a lot of modded items use that as an identifier.